aiida/schedulers/plugins/direct.py

```diff
@@ -55,6 +55,7 @@
     'W': JobState.RUNNING,
     'X': JobState.DONE,
     'Z': JobState.DONE,
+    '?': JobState.UNDETERMINED,
```

Comment:
Fine to add this, but could you please add a comment that this is specifically for MacOS (maybe with a link to a resource where you found the information) happy to merge this

Reply:
Actually, finding any information on this is quite tricky to be honest. Originally I got my answer from ChatGPT. ^^ [This answer](https://apple.stackexchange.com/a/460403/497071) on apple.stackexchange seems quite reasonable, though.

Turns out `?` is the default state returned by `ps`. Seems Apple must have modified the `ps` command and failed to catch a certain state, which means it will return the `?` default:

https://github.com/freebsd/freebsd-src/blob/main/bin/ps/print.c#L262

On the train now, but will add a comment once I have access to my `core` dev environment again.